prime-router/src/test/kotlin/fhirengine/azure/FHIRRouterIntegrationTests.kt

```diff
@@ -0,0 +1,1127 @@
+                )
+            }
+
+            // make sure action table has a new entry
+            checkActionTable(listOf(TaskAction.convert, TaskAction.receive, TaskAction.route))
+
+            // ACTION_LOG table is expected to have a record with no record_id identifying an observation filter action
+            // that may or may not have happened.
+            checkActionLogTable(expectedButStrangeObservationFilterEntry)
+        }
+    }
+
+    @Test
+    fun `should send valid FHIR report filtered by condition code 94558-5`() {
+        // set up
+        val reportContents =
+            listOf(
+                File(MULTIPLE_OBSERVATIONS_FHIR_URL).readText()
+            ).joinToString()
+
+        val reportPair = createReportsWithLineage(reportContents)
+        val receiveReport = reportPair.first
+        val convertReport = reportPair.second
+        val queueMessage = generateQueueMessage(convertReport, reportContents, fhirSenderWithNoTransform)
+        val fhirFunctions = createFHIRFunctionsInstance()
+
+        // make sure action table has only what we put in there
+        checkActionTable(listOf(TaskAction.convert, TaskAction.receive))
+
+        // execute
+        var receiverSetupData = listOf(
+            ReceiverSetupData("x", conditionFilter = observationFilter),
+            ReceiverSetupData("y"),
+        )
+        var receivers = createReceivers(receiverSetupData)
+        val org = createOrganizationWithReceivers(receivers)
+        val fhirRouter = createFHIRRouter(org)
+        fhirFunctions.doRoute(queueMessage, 1, fhirRouter)
+
+        // check results
+        ReportStreamTestDatabaseContainer.testDatabaseAccess.transact { txn ->
+            // did the report get pushed to blob store correctly and intact?
+            val routedReports = verifyLineageAndFetchCreatedReportFiles(convertReport, receiveReport, txn, 2)
+            val reportAndBundles =
+                routedReports.map {
+                    Pair(
+                        it,
+                        BlobAccess.downloadBlobAsByteArray(it.bodyUrl, getBlobContainerMetadata())
+                    )
+                }
+
+            var fhirBundlesAsObjectsOnly = reportAndBundles.map { it.second.toString(Charsets.UTF_8) }
+                                                           .map { FhirTranscoder.decode(it) }
+
+            var fhirBundleReceiverX = fhirBundlesAsObjectsOnly[0]
+            var fhirBundleReceiverY = fhirBundlesAsObjectsOnly[1]
+
+            // there should only be one observation of five remaining, and the code of that observation
+            // should be 94558-5
+            assertEquals(1, fhirBundleReceiverX.getObservations().size)
+            assertEquals(1, fhirBundleReceiverX.getObservations()[0].code.coding.size)
+            assertThat(fhirBundleReceiverX.getObservations()[0].code.coding[0].code.equals("94558-5"))
+
+            // for receiver Y all five observations should be intact
```

Comment:
Where are we verifying the "orphaned resources" that get pruned when an observation get removed? If we aren't yet, that could be a new ticket since "orphan pruning" logic is lacking product reqs. see https://github.com/CDCgov/prime-reportstream/issues/12888.

Reply:
we are not verifying orphaned resource pruning. I did so by hand yesterday and added the results to the one sheet describing the defects these tests discovered. happy to make a new ticket.

Reply:
banged all of this out in one go. Happy to change it as per your request 

https://app.zenhub.com/workspaces/platform-6182b02547c1130010f459db/issues/gh/cdcgov/prime-reportstream/14568

Reply:
Great, thank you! Just to be clear, this is blocked on product requirements regarding what actually should be pruned or not.